### ISSUE TITLE
Regular Expression Denial of Service (ReDoS)

### ISSUE THREAD

**Issue by phongle2512** [medikoo/memoizee#146]
Snyk reports Regular Expression Denial of Service (ReDoS) on memoizee@0.4.17

<img width="1179" alt="Screenshot 2024-12-06 at 15 10 33" src="https://github.com/user-attachments/assets/a211e66e-c785-40f7-b690-cb9373475fc7">

**Comment by medikoo** [comment]
@phongle2512 screenhost didn't upload. Can you provide more details, maybe a link to Snyk page?

**Comment by phongle2512** [comment]
> @phongle2512 screenhost didn't upload. Can you provide more details, maybe a link to Snyk page?

@medikoo Re-attach the screenshot
Detailed paths
Introduced through: memoizee@0.4.17 › d@1.0.2 › es5-ext@0.10.64 › es6-iterator@2.0.3 › d@1.0.1 › es5-ext@0.10.61
Security information
Factors contributing to the scoring:
Snyk: [CVSS 7.5](https://security.snyk.io/vuln/SNYK-JS-ES5EXT-6095076) - High Severity
NVD: Not available. NVD has not yet published its analysis.

**Comment by medikoo** [comment]
@phongle2512 this report is incorrect.

The corresponding vulnerability was fixed with `es5-ext@0.10.63`, while `memoizee@0.4.17` at minimum requires `es5-ext@0.10.64`

**Comment by phongle2512** [comment]
@medikoo You can see the detailed paths that Synk was reported
Detailed paths
Introduced through: memoizee@0.4.17 › d@1.0.2 › es5-ext@0.10.64 › es6-iterator@2.0.3 › d@1.0.1 › es5-ext@0.10.61

**Comment by medikoo** [comment]
Yes, but this not how this will be ever installed, as `memoizee` requires at least `es5-ext@0.10.64`, all its other dependencies will also depend on this version

There's no package manager that will install it differently, feel free to test

**Comment by phongle2512** [comment]
From the detailed paths, Synk is reporting to your es6-iterator@2.0.3 › d@1.0.1 › es5-ext@0.10.61, it should use the d@1.0.2 instead.

**Comment by medikoo** [comment]
>From the detailed paths, Synk is reporting to your es6-iterator@2.0.3 › d@1.0.1 › es5-ext@0.10.61, it should use the d@1.0.2 instead.

`es-iterator`'s dependency on `d` is [`^1.0.1`](https://github.com/medikoo/es6-iterator/blob/049b1ef04bae3f38142b676604639f3d363ef722/package.json#L19), which means at minimum v1.0.1. If anything requires any greater version in v1 range, it'll be used instead (not to mention, that installers always install the latest that match the range). Here `memoizee` requires `^1.0.2`, so it's what `es6-iterator` ends with.

Feel free to play with any node packages installer when installing `memoizee`, and you'll never see `es5-ext` below v0.10.64` installed beneath
